### ISSUE TITLE
registering resource named "Controller" gives issues

### ISSUE THREAD

**Issue by pinkynrg** [activeadmin/activeadmin#4908]
Hello everyone and thanks for this marvelous gem!

I've been trying to register a resource unfortunately called "Controller" and I'm having some issues. Would like to remark that I do understand that this is a reserved word of some sort. 

This morning I wasn't able to see neither show/edit page. After registering that resource with an alias HostController (using the `as` parameter), show/edit page were available. 

Now I'm having another issue and I'm quite stuck with it. 

It seems that the form is getting populated with fields still named like 

- controller[name]
- controller[identifier]
- controller[another_resource_id]

and so on. 

Once I submit the form nothing is done. I byebug-ed and checked the update method and it seems that params got stripped by the `controller` payload. I did another test where I forced the name of the fields to `ctrl` (using the Developer Tools), and I was then able to byebug and see that the `ctrl` payload was present. 

The first thing in my mind would be to rename `controller` -> `host_controller` also in the form and then use that in the controller methods create/update.

Is this the right approach? What would be the ideal way to do this? I tried to customize the form and I'm having a hard time. A small hint would be appreciated.

Thanks again for this awesome gem.

**Comment by varyonic** [comment]
Yes.  Maybe create an empty subclass of Controller with a different name and register that.

**Comment by pinkynrg** [comment]
Interesting approach I will try that and I will let you know.

Thanks

**Comment by pinkynrg** [comment]
just tried and it works nicely. 

Before accepting this as final solution I was wondering if it was going to be too much of an hassle to override the standard form in order to have things work without creating an extra subclass.

Thanks for your patience

**Comment by varyonic** [comment]
ActiveAdmin::FormBuilder?  IMHO not worth it but the code if there if you want to try.

**Comment by pinkynrg** [comment]
Thanks i will proceed
